src/chttpd/src/chttpd_util.erl

```diff
@@ -95,3 +99,68 @@ handle_legacy_config(OriginalHeaders, DefaultHeaderValue) ->
         false ->
             OriginalHeaders
     end.
+
+handle_request_helper(Req) ->
+    Args = couch_mrview_http:parse_params(Req, undefined),
+    ShardDbName = config:get("mem3", "shards_db", "_dbs"),
+    %% shard_db is not sharded but mem3:shards treats it as an edge case
+    %% so it can be pushed thru fabric
+    {ok, Info} = fabric:get_db_info(ShardDbName),
+    Etag = couch_httpd:make_etag({Info}),
+    Options = [{user_ctx, Req#httpd.user_ctx}],
+    {ok, Resp} = chttpd:etag_respond(Req, Etag, fun() ->
+        {ok,Resp}=chttpd:start_delayed_json_response(Req,200,[{"ETag",Etag}]),
+        VAcc = #vacc{req=Req,resp=Resp},
+        fabric:all_docs(ShardDbName, Options, fun callback_func/2, VAcc, Args)
+    end),
+    case is_record(Resp, vacc) of
+        true -> {ok, Resp#vacc.resp};
+        _ -> {ok, Resp}
+    end.
+
+callback_func({meta, _Meta}, #vacc{resp=Resp0}=Acc) ->
```

Comment:
`callback_func` is also a rather generic name. It may work in a module which handles only one request type and has only one obvious callback but say, if we added another helper here which needed a callback and used the same pattern, we'd end up with `callback_func2` or something of that sort

Reply:
Change to `all_dbs_or_dbs_info_callback`